src/python/platforms/fuchsia/util/device.py

```diff
@@ -282,7 +289,18 @@ def process_logs(self, logfile, guess_pid=False):
 
           # Echo the line
           tmp.write(line)
-    os.rename(logfile + '.tmp', logfile)
+
+    # Clusterfuzz's stack analyzer expects the
+    # `==[num]== ERROR: [SanitizerName]: [failure type]` line
+    # to occur *before* the stacktrace, so make a new tempfile
+    # where we insert that line at the top.
+    with open(logfile + '.tmp', 'r+') as tmp:
+      with open(logfile + '.tmp2', 'w') as tmp2:
+        tmp2.write(line_with_crash_message)
+        for line in tmp:
```

Comment:
rather than iterate line by line here, we can just use `shutil.copyfileobj`

Reply:
i don't think this works, though?  shutil.copyfileobj clobbers the whole file with the source, and we need that first line to stay there... if there's a better way to append one file to another, though, while still keeping that first line there, i'm all ears

Reply:
I believe shutil will still work here, as it'll respect the current position (which has advanced because of your `write()` call).

Reply:
good point, fixed